subsys/usb/host/usbh_shell.c

```diff
@@ -0,0 +1,541 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdlib.h>
+#include <string.h>
+#include <errno.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/shell/shell.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/usb/usbh.h>
+
+#include "usbh_internal.h"
+#include "usbh_ch9.h"
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(usbh_shell, CONFIG_USBH_LOG_LEVEL);
+
+#define FOOBAZ_VREQ_OUT		0x5b
+#define FOOBAZ_VREQ_IN		0x5c
+
+USBH_CONROLLER_DEFINE(uhs_ctx, DEVICE_DT_GET(DT_NODELABEL(zephyr_uhc0)));
+
+const static struct shell *ctx_shell;
+
+static void print_dev_desc(const struct shell *sh,
+			   const struct usb_device_descriptor *const desc)
```

Comment:
This function is called with byte buffer, not struct itself. Therefore, the appropriate endian swapping (with `sys_le16_to_cpu`) has to be done here before printing.